altair/utils/core.py

```diff
@@ -436,7 +437,7 @@ def sanitize_arrow_table(pa_table):
 
 def parse_shorthand(
     shorthand: Union[Dict[str, Any], str],
-    data: Optional[pd.DataFrame] = None,
+    data: Optional[_DataFrameLike] = None,
```

Comment:
I'm not sure, but this a good callout. This type should be `Union[_DataFrameLike, pd.DataFrame]` anyway since it supports pandas DataFrame's that don't have `__dataframe__` methods.

Reply:
Signatures updated in [c4a9b08](https://github.com/altair-viz/altair/pull/3114/commits/c4a9b08880ba4fa1951cfbb2be98f0ec28485cc2)